src/cracking.py

```diff
@@ -0,0 +1,307 @@
+
+# This is an example input
+
+## Would you rather I print out the output in hashsearch.py or in here ##
+
+class Searcher:
+
+	def __init__(self, config):
+		self.searchers_offline = [] # Temp disabled until I get the IDs format
+		self.searchers_online = [hashtoolkit(), hashsorg(), LmRainbowTabels(), nitrxgen()] # md5cryp() takes ~ 3 seconds to do
+		self.Hash_input = namedtuple("Hash_input", ["text", "types", "api_keys"])
+		
+		self.perform_search(config)
+	
+	def perform_search(self, config):
+
+		for hash in config["hashes"]:
+
+			hash_ctext = next(iter(hash.keys()))
+			keys = [x.lower() for x in hash[next(iter(hash.keys()))].keys()]
+			supported_searchers = []
+			types = []
+
+			if not config["offline"]:
+				for search in self.searchers_online:
+					for hashtype in keys:
+						if hashtype in search.supports:
+							supported_searchers.append(search)
+							types.append(hashtype)
+							break
+
+			for search in self.searchers_offline:
+				for hashtype in keys:
+					if hashtype in search.supports:
+						supported_searchers.append(search)
+						types.append(hashtype)
+						break
+
+			future = self.Hash_input(hash_ctext, types, config["api_keys"])
+
+			with console.status(f"[bold magenta] Searching [reversed] {hash_ctext}") as status:

```

Comment:
Use the colour scheme from here if ya want!
https://github.com/HashPals/Name-That-Hash/wiki/Accessibility